tools/testbench/topology.c

```diff
@@ -729,6 +740,10 @@ int parse_topology(struct sof *sof, struct shared_lib_table *library_table,
 
 			debug_print(message);
 
+			/* update max pipeline_id */
+			if (hdr->index > tp->max_pipeline_id)
```

Comment:
why greater than? Are we always walking forward this way?

Reply:
The reason of recording the max pipeline ID is for walking through all pipelines by ID (from 1 to max_pipeline_id) in the future. I think it is a common convention that designers add pipelines with monotonic increasing pipeline IDs so we could fulfill the walk by this simple way.

Moreover, because this is the testbench, we could regulate the design behavior of the test typologies. Fort this reason I think recording the max pipeline ID is enough for the walk.